sdk/src/main/java/io/radar/sdk/RadarSettings.kt

```diff
@@ -311,15 +314,49 @@ internal object RadarSettings {
         return RadarFeatureSettings.fromJson(JSONObject(optionsJson))
     }
 
+    fun setSdkConfiguration(context: Context, configuration: RadarSdkConfiguration?) {
+        Radar.logger.d("set SDK Configuration | sdkConfiguration = $configuration")
+        if (configuration != null) {
+            getSharedPreferences(context).edit { putInt(KEY_LOG_LEVEL, configuration.logLevel.value) }
+
+            val sdkConfigurationString = configuration.toJson().toString()
+            getSharedPreferences(context).edit { putString(KEY_SDK_CONFIGURATION, sdkConfigurationString) }
+        }
+    }
+
+    fun getSdkConfiguration(context: Context): RadarSdkConfiguration? {
+        val sdkConfigurationString = getSharedPreferences(context).getString(KEY_SDK_CONFIGURATION, null)
+        return RadarSdkConfiguration.fromJson(sdkConfigurationString?.let { JSONObject(it) })
+    }
+
+    fun getClientSdkConfiguration(context: Context): JSONObject {
+        val sharedPrefClientSdkConfig = getSharedPreferences(context).getString(KEY_CLIENT_SDK_CONFIGURATION, null);
+
+        return if (sharedPrefClientSdkConfig != null) {
+            JSONObject(sharedPrefClientSdkConfig)
+        } else {
+            JSONObject()
+        }
+    }
+
     internal fun getLogLevel(context: Context): Radar.RadarLogLevel {
         val logLevelInt = getSharedPreferences(context).getInt(KEY_LOG_LEVEL, 3)
         val userDebug = getUserDebug(context)
         return if (userDebug) Radar.RadarLogLevel.DEBUG else Radar.RadarLogLevel.fromInt(logLevelInt)
     }
 
     internal fun setLogLevel(context: Context, level: Radar.RadarLogLevel) {
-        val logLevelInt = level.value
-        getSharedPreferences(context).edit { putInt(KEY_LOG_LEVEL, logLevelInt) }
+        val sdkConfiguration = getClientSdkConfiguration(context)
```

Comment:
I think we should keep `RadarSettings` pure and move this "syncing" logic into a wrapper that lives in `RadarLogger`